netsim/quirks/eos.py

```diff
@@ -24,11 +24,28 @@ def check_mlps_vlan_bundle(node: Box) -> None:
     node.interfaces = [ intf for intf in node.interfaces if intf.ifname != ifname ]
 
 def check_mpls_clab(node: Box, topology: Box) -> None:
-  if topology.provider == 'clab':
-      common.error(
-        f'Arista cEOS ({node.name}) does not support MPLS. Use vEOS VM with libvirt provider',
-        common.IncorrectType,
-        'quirks')
+  if devices.get_provider(node,topology) == 'clab':
+    common.error(
+      f'Arista cEOS ({node.name}) does not support MPLS. Use vEOS VM with libvirt provider',
+      common.IncorrectType,
+      'quirks')
+
+def check_shared_mac(node: Box, topology: Box) -> None:
+  if devices.get_provider(node,topology) != 'clab':
+    return
+
+  for intf in node.interfaces:
+    if get_from_box(intf,'gateway.protocol') != 'anycast':              # We hope that VRRP works (not tested yet)
+      continue
+
+    if get_from_box(intf,'vlan'):                                       # Anycast works on VLAN cEOS interfaces
+      continue
+
+    common.error(
```

Comment:
This is also true for SR Linux - anycast only works on IRB interfaces (not bridge, not routed interface)

Reply:
So you'll have to add a device quirk ;) I'm not adding a feature flag (I learned my lessons with VLAN module).

NXOS is even worse -- looks like anycast gateway works only with EVPN.